vortex-buffer/src/buffer.rs

```diff
@@ -10,14 +12,32 @@ use crate::debug::TruncatedDebug;
 use crate::{Alignment, BufferMut, ByteBuffer};
 
 /// An immutable buffer of items of `T`.
-#[derive(Clone, PartialEq, Eq, PartialOrd, Hash)]
+#[derive(Clone, Eq)]
 pub struct Buffer<T> {
     pub(crate) bytes: Bytes,
     pub(crate) length: usize,
     pub(crate) alignment: Alignment,
     pub(crate) _marker: std::marker::PhantomData<T>,
 }
 
+impl<T> PartialEq for Buffer<T> {
+    fn eq(&self, other: &Self) -> bool {
+        self.bytes == other.bytes
```

Comment:
I don't really know what the Rust convention is here. Should the default compare everything (including alignment) and an extra function `eq_bytes` ignores it? Or the other way around?

Reply:
I think the standard way is that `PartialEq` is the semantic equality, and sometimes people add "structural" equality which is a dedicated fn.